waspc/src/Wasp/Node/Version.hs

```diff
@@ -1,42 +1,63 @@
 module Wasp.Node.Version
-  ( getAndCheckNodeVersion,
-    getNodeVersion,
-    nodeVersionRange,
-    latestMajorNodeVersion,
-    waspNodeRequirementMessage,
-    makeNodeVersionMismatchMessage,
+  ( getAndCheckUserNodeVersion,
+    VersionCheckResult (..),
+    oldestWaspSupportedNodeVersion,
+    parseNodeVersionOutput,
   )
 where
 
+import Control.Arrow (left)
 import Data.Conduit.Process.Typed (ExitCode (..))
+import Data.Function ((&))
 import System.IO.Error (catchIOError, isDoesNotExistError)
 import qualified System.Process as P
-import Text.Read (readMaybe)
-import qualified Text.Regex.TDFA as R
+import qualified Text.Parsec as P
 import qualified Wasp.SemanticVersion as SV
+import qualified Wasp.SemanticVersion.Version as SV
 import Wasp.Util (indent)
 
--- | Gets the installed node version, if any is installed, and checks that it
+-- | Wasp supports any node version equal or greater to this version.
+-- | We usually keep this one equal to the latest LTS.
+-- NOTE: If you change this value, make sure to also update it on some other places:
+--   - /.github/workflows/ci.yaml -> actions/setup-node -> node-version
+--   - /waspc/.nvmrc
+--   - /web/docs/introduction/getting-started.md -> "Requirements" section.
+oldestWaspSupportedNodeVersion :: SV.Version
+oldestWaspSupportedNodeVersion = SV.Version 18 0 0
+
+data VersionCheckResult
+  = VersionCheckFail !ErrorMessage
+  | VersionCheckSuccess !(Maybe WarningMessage) !SV.Version
+
+type WarningMessage = String
+
+type ErrorMessage = String
+
+-- | Gets the user's installed node version, if any is installed, and checks that it
 -- meets Wasp's version requirement.
---
--- Returns a string representing the error
--- condition if node's version could not be found or if the version does not
--- meet the requirements.
-getAndCheckNodeVersion :: IO (Either String SV.Version)
-getAndCheckNodeVersion =
-  getNodeVersion >>= \case
-    Left errorMsg -> return $ Left errorMsg
-    Right nodeVersion ->
-      if SV.isVersionInRange nodeVersion nodeVersionRange
-        then return $ Right nodeVersion
-        else return $ Left $ makeNodeVersionMismatchMessage nodeVersion
+getAndCheckUserNodeVersion :: IO VersionCheckResult
+getAndCheckUserNodeVersion =
+  getUserNodeVersion >>= \case
+    Left errorMsg -> return $ VersionCheckFail errorMsg
+    Right userNodeVersion ->
+      return $
+        if SV.isVersionInRange userNodeVersion $ SV.Range [SV.gte oldestWaspSupportedNodeVersion]
+          then VersionCheckSuccess Nothing userNodeVersion
+          else VersionCheckFail $ makeNodeVersionMismatchMessage userNodeVersion
+
+makeNodeVersionMismatchMessage :: SV.Version -> String
+makeNodeVersionMismatchMessage nodeVersion =
+  unlines
+    [ "Your Node version does not meet Wasp's requirements! You are running Node " <> show nodeVersion <> " .",
```

Comment:
Correct. I wanted it to be clear that dot is not part of the version number, otherwise it can look like as if version number possibly got cut off in the middle of it.

Reply:
I'd say the space before the full stop makes the output look more broken. I know it confused me when I ran it.

How about we ditch the full stop altogether?

Reply:
The problem is that next sentence comes after it, so if we ditch it, it is weird.
Let's just leave it.

Reply:
Ok, I ended up removing the extra space!